### PR TITLE
develop/implement-filterOpenCoffeeShops-in-curatedlist

### PR DESCRIPTION
**Pull request by jeongJ2004**
This PR introduces a new feature to dynamically filter coffee shops based on their current open hours and enhances test coverage for this functionality. It also includes necessary build configuration updates to support the new tests.

Key Changes :
- Feature : Filtering Open Coffee Shops
   1. Added the filterOpenCoffeeShops function to dynamically filter coffee shops based on their current open hours.
   2. Ensures proper handling of invalid or missing time information in the filtering process.
   3. Allows seamless integration with existing functionalities, such as fetchAndSortCoffeeShopsByRating.

- Unit Tests for Filtering Open Coffee Shops :
   1. Introduced unit tests for filterOpenCoffeeShops to validate correct filtering behavior.
   2. Added tests to handle edge cases, including invalid or missing time information.
   3. Mocked LocalTime.now() to simulate various time scenarios for accurate and consistent tests.
   4. Verified compatibility with fetchAndSortCoffeeShopsByRating to ensure existing functionality remains intact.

- Build Configuration Enhancements :
   1. Added the --add-opens JVM option in build.gradle.kts to enable reflective access during unit testing.
   2. Ensures that the test suite runs successfully in all environments by resolving compatibility issues.

Details for Unit Tests :
- Validated that coffee shops are correctly filtered based on opening and closing hours.
- Confirmed that coffee shops with invalid or missing time data are excluded from the filtered results.
- Ensured that existing sorting functionality (fetchAndSortCoffeeShopsByRating) works seamlessly alongside the new filtering feature.

### CODE REVIEW

File: app/src/main/java/com/android/brewr/model/coffee/CuratedCoffeeShopList.kt [hunk 1]
```diff
@@ -52,3 +54,29 @@ fun fetchAndSortCoffeeShopsByRating(
...
+ * Filters a list of coffee shops to include only those that are currently open based on their
+ * opening and closing hours.
+ *
+ * @param coffeeShops A [List] of [Coffee] objects representing coffee shops. Each [Coffee] object
+ *   contains details such as its name, location, and a list of [Hours] objects. The [Hours] objects
+ *   specify the opening and closing times of the coffee shop.
+ * @return A [List] of [Coffee] objects that are currently open. Coffee shops with invalid or
+ *   missing time information are excluded.
+ */
+fun filterOpenCoffeeShops(coffeeShops: List<Coffee>): List<Coffee> {
```
Comment (LeiChen5) [inline app/src/main/java/com/android/brewr/model/coffee/CuratedCoffeeShopList.kt:68]: It's a really good idea to filter coffee shops that are open

**Review by LeiChen5 — APPROVED** [review]
LGTM! Line coverage is above 80% good job!
